src/coord/src/client.rs

```diff
@@ -527,8 +499,15 @@ impl SessionClient {
 
 impl Drop for SessionClient {
     fn drop(&mut self) {
-        if self.session.is_some() {
-            panic!("unterminated SessionClient dropped")
+        // We may not have a session if this client was dropped while awaiting
+        // a response. In this case, it is the coordinator's responsibility to
+        // terminate the session.
+        if let Some(session) = self.session.take() {
+            self.inner
+                .inner
+                .cmd_tx
+                .send(Command::Terminate { session })
```

Comment:
Could we not do this before because something was async but isn't anymore?

Reply:
I sure hope that’s the reason, yeah! I didn’t bother to check. I could believe that this was always possible and we just had no idea.